update to python 3

ineeee opened the pull request:
i updated taigabot to python 3 and confirmed it works

i havent finished testing all of the plugins

Commit:
experimental: use a named pipe to handle irc messages in py2 and py3

Commit:
py3 experiment: clean a bit

Commit:
2to3: part 1

Commit:
2to3: part 2 (it runsgit status)

Commit:
2 to 3: some plugins

I NEED SOME WEED IN MY LIFE RIGHT NOW AAAAAAAAAAAAAAA>>FGDVZ

Commit:
fuck this shit it belongs on another branch

Commit:
2 to 3: i have a headache

Commit:
2 to 3: idk tbh

Commit:
2 to 3: update print function in plugins

Commit:
2 to 3: update some plugins

Commit:
2 to 3: automatically upgrade plugins, didnt test shit

Commit:
2 to 3: update last two utils

Commit:
cherry-pick: core changes idk i forgot

Commit:
2 to 3: i forgot this one file

Commit:
2 to 3 updates

Commit:
type hinting

Commit:
type hint the bot core

Commit:
type hint the bot core

Commit:
clean strings a bit

Commit:
clean stdout a bit

ineeee commented:
im going to ping the fuck out of everyoen who has contributed lately cuz i need reviewers

@blompf @adedomin @676339784 @eskimo @jesseerwin @DemonOnFishy @a1k1os @nojusr

Commit:
core cleanup (also chmod -x)

Commit:
this was bugging me

Commit:
remove encoding hacks from py2

Commit:
update plugins to python 3

tested and working: fmylife.py duckduckgo.py distro.py distance.py dictionary.py dice.py debt.py cypher.py coins.py coin.py choose.py calculator.py booru.py bash.py bank.py

DemonOnFishy commented on bot.py:
```diff
@@ -1,63 +1,62 @@
+#               |___/     irc bot
+
+__author__ = 'inexistence'
+__credits__ = ['infinity', 'thenoodle', '_frozen', 'rmmh', '676339784', 'adedomin', 'eskimo', 'blompf']
```
![yfw contributed to daddiescummies](https://ih1.redbubble.net/image.2854641226.5367/st,small,845x845-pad,1000x1000,f8f8f8.jpg)

ineeee replied:
sorry about that guints, i added u in https://github.com/inexist3nce/Taigabot/commit/53302b8d5dd08773f75afaed1696f371d079001f

i already tested the bot in irc, i'm currently going through plugins testing every single command but that takes a while cuz many have `u""`/encode/decode hacks to deal with python 2 so i have to undo that before they run

DemonOnFishy reviewed (approved):
LBTM (looks based to me)

Not sure if these changes were deployed and tested in IRC directly, but LMK if you want any assistance with testing

Commit:
add contributors in roughly chrono order

adedomin commented on plugins/py3test.py:
```diff
@@ -0,0 +1,92 @@
+# so data transfers should be pretty much instant with no cpu usage
+
+# TODO find a better way to encode taiga data in a line
+# just went with 'unicode unit separator' because its unlikely to be in an irc message
```
NOOOOOOOOOOOOOOOOOOOOOOOOoooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooo

adedomin replied:
we should probably use a Tag-Length-Value or some other common document format, like god forbid JSON. hold up on this. let me fuck with it locally.

ineeee replied:
i think i'm going to leave this thing to the other branch, maybe even add the stdin/stdout thing for adding neo8ball-like plugin support

ineeee replied:
### **dude lets use xml**

adedomin commented:
i'd say take the 2/3 pipe stuff out and merge the rest.

adedomin reviewed (changes requested):
sure... I still think the py3test/bot thing should be split off.

Commit:
update plugins to py 3

checked: kernel.py lastfm.py furry.py gelbooru.py geoip.py google.py hashtags.py heartbleed.py lyrics.py

Commit:
bump requirements

Commit:
fix crash when plugins send bytes instead of strings

Commit:
fix next() raising stopiteration in py3

Commit:
update more plugins to py3

working: radio.py religion.py mediawiki.py marketing.py quote.py
others: amazon.py masshighlight.py wolframalpha.py

Commit:
add flake8 guidelines

Commit:
why the fuck are files executable by default UGH

Commit:
more tested plugins, update to py3, bump requirements

Commit:
update requirements, document a bit where they're used

ineeee commented:
currently the bot works really well in python 3.9.7 in an ubuntu container. so far i've tested 40 plugins (34 ok, 6 broken) and only 34 plugins remain.

i updated the dependencies to roughly the latest versions, we now run lxml 4.4, bs 4.9, requests 2.27 (lxml is still a pain in the ass to install though, still needs a compiler)

Commit:
update more plugins

ok: core_admin_global, core_ctcp, datafiles, help
broken: core_admin_channel, major_league_baseball
cant test: core_misc, core_sieve, masshighlight

Commit:
update more plugins

ok: nfl, twitter, utility, validate, vimeo, weather, wordoftheday, youtube

Commit:
fix weird/unknown database encode error

Commit:
update more plugins

ok: spotify, stock, stupid, times, urls

ineeee commented:
## plugin progress
updated to 83da92397ef5936bc3ef601ef5603d91672190d7

### unknown
- newegg.py

### broken (help needed)
- amazon.py: amazon changed html, scraper broke
- major_league_baseball.py: the api was deprecated, unrelated to python 3
- media.py: what the fuck is StringIO(initial_value)
- metacritic.py: scraper broke, metacritic changed html
- steam.py: scraper broke, steam changed htm


### working
- 4chan.py: all works except regex search
- ai.py
- bank.py
- bash.py
- booru.py
- calculator.py
- choose.py
- coin.py
- coins.py
- core_admin_global.py: all works except .set
- core_ctcp.py: all works except .host
- core_user.py: all works except horoscope scraping
- cypher.py
- datafiles.py
- debt.py
- dice.py
- dictionary.py
- distance.py
- distro.py
- duckduckgo.py
- fmylife.py
- furry.py
- geoip.py
- google.py
- hashtags.py
- heartbleed.py
- help.py
- kernel.py
- lastfm.py
- log.py
- lyrics.py
- marketing.py
- mediawiki.py
- namegen.py
- nfl.py: i saw adrift test it
- radio.py
- religion.py
- reminder.py
- seen.py
- spellcheck.py
- spotify.py
- stock.py
- stupid.py: all works (i think cash stuff has problems: bet, sell, mug, donate, pay, owed)
- system.py: did not test windows code
- tell.py: it works but i didnt test or clean it
- times.py
- todo.py
- translate.py
- twitch.py
- twitter.py
- urbandict.py
- urls.py: random page/image links work, but didnt test thoroughly
- utility.py: works ok, base64/unicode stuff broken
- validate.py
- vimeo.py
- weather.py
- wordoftheday.py
- youtube.py: all works, didnt test playlist titles
- core_admin_channel.py
- countdown.py
- gelbooru.py: randomly returns "Not found", i think it isnt taiga/py3 related
- quote.py: `.qchan` is buggy
- wolframalpha.py

### cant test
- core_misc: seems to be ok
- core_sieve
- masshighlight
- mygengo_translate: i dont know how to get the mygengo package working
- soundcloud

so far only adrift says to hold on to merging this branch.

Commit:
finish updating plugins

working: 4chan, ai, core_user, log, namegen, reminder, todo. not working: metacritic, mygengo, soundcloud, steam

Commit:
update requirements

ineeee commented:
hey im done updating the bot and all of its plugins. i tested every single command by hand.

i dont feel like dealing with the 9 broken plugins so that's going to be an issue for someone else in the future (probably me lol)

Commit:
run git merge

ineeee reviewed (commented):
what do reviews do if im the pr author????

Commit:
fix stuff broken by merge

676339784 committed:
Fix countdown bug

Commit:
Merge pull request #24 from 676339784/patch-3

Fix countdown bug

Commit:
move loose variables to Bot class

Commit:
bot.py: catch ctrl+c to quit

Commit:
upgrade lxml to support py 3.10

Commit:
add support for python 3.10 (and install isntructions)

ineeee commented:
taigabot is now capable of running on python 3.9 and 3.10. i've included ubuntu/debian install instructions.

Commit:
countdown fixed in b0b15b395c5a2b45b37dbd46bfbcff1c59dc7337

Commit:
move Bot class to its own file in core folder

Commit:
fix wolfram alpha

Commit:
core_admin_channel, core_misc: fix .admin, .autoop

Commit:
core_admin_channel: fix ignore/unignore

also my editor replaced the quotes

Commit:
delete pyexec: api is offline, eval is bad

Commit:
http.py: delete unused methods, py2 encode stuff

checked whole codebase with ripgrep

Commit:
database.py: type hinting

Commit:
chmod -x plugins/util/*.py

i still dont know why all of taigabot files are 755d

Commit:
core: move load_config to bot class

also crlf -> lf

Commit:
delete stock.py: we have a new stock.py and yahoo api is dead

Commit:
mylifeisaverage: the page went offline in 2019

Commit:
answers.py: yahoo answers died

Commit:
gitio.py: service went read-only, cant submit urls

Commit:
oblique: service has been offline for 11 years

the appengine service is offline. oblique was used by the phenny bot
(a very old skybot/uguubot fork)
- original website was http://wiki.github.com/nslater/oblique/
- last known repo is https://github.com/clsn/oblique
- last commit is 2011

Commit:
title.py: superseded by urls.py

Commit:
request.py: superseded by tell.py

Commit:
fuck downloading posted urls, who thought that was a good idea? also they're passed through subprocess.Popen??? bruh

ineeee commented:
```
<adrift> yea go ahead and merge
```
![shitpost.jpg](https://user-images.githubusercontent.com/10874672/172973051-c29104e6-0e60-4103-a60a-018d869a8b4f.jpg)

Commit:
Merge branch 'dev' into breakc
